SofaKernel/framework/sofa/helper/io/Image.cpp

```diff
@@ -411,11 +411,28 @@ bool Image::save(std::string filename, int compression_level)
 
 Image* Image::Create(std::string filename)
 {
-    std::string loader="default";
+    std::string extension="default";
     std::string::size_type p = filename.rfind('.');
     if (p!=std::string::npos)
-        loader = std::string(filename, p+1);
-    return FactoryImage::CreateObject(loader, filename);
+        extension = std::string(filename, p+1);
+    Image* createdImage = FactoryImage::CreateObject(extension, filename);
+    if( extension.compare("default")>0 )
+    {
+        if(!createdImage )
+        {
+            helper::vector<std::string> validExtensions;
+            helper::io::Image::FactoryImage::getInstance()->uniqueKeys(std::back_inserter(validExtensions));
+            msg_error("Image") << "Could not load image with extension " << extension << ". Valid extensions: " << validExtensions;
+        }
+    }
+    else
+    {
+        helper::vector<std::string> validExtensions;
+        helper::io::Image::FactoryImage::getInstance()->uniqueKeys(std::back_inserter(validExtensions));
+        msg_error("Image") << "No extension detected. Valid extensions: " << validExtensions;
```

Comment:
Tips of the day: 
If you have a lot of msg_error("xxx") you can use the macro 
MSG_REGISTER_CLASS(....)
After using it you can just do msg_info() << 
search example of how to use it in the sofa source code.

Reply:
There is no tricks, the macro is just a plain gold old macro that replace the macro with a longer text. I don't understand the connection you make with forward definition and not having the macro used in the .cpp.  But you can find a lot of uses of the macro in the code base (eg: SphereLoader.cpp, meshTrian.cpp, ...)

I definitively support anyone writing the corresponding docs in the messaging documentation :)